### PR TITLE
only enable latest tag for docker if build tag equals latest github tag

### LINKED ISSUE #2167: Docker Hub tags are stale
The generalized and latest tags (e.g. `v5` and `latest`) are pointed to the last release, not the highest released version. Thus, if `v5.3.0` gets released and then bugfix version `v5.2.9`, the latter is pointed to by the `v5` and `latest`. But the should point to the highest released version, meaning `v5.3.0`.

### PR DESCRIPTION
**Pull request by woutslakhorst**
fixes #2167 

`latest` was always created, now it's only enabled if the tag being build is the latest github tag.

### CODE REVIEW

**Review by reinkrul — APPROVED** [review]

File: .github/workflows/build-images.yaml [hunk 1]
```diff
@@ -45,7 +52,7 @@ jobs:
             # generate v5 tag
             type=semver,pattern={{major}}
           flavor: |
-            latest=auto
+            latest=${{ github.ref == steps.get-latest-tag.outputs.tag }}
```
Comment (reinkrul) [inline .github/workflows/build-images.yaml:55]: I find it hard to understand, but `latest` should be either `true`, `false`, or `auto` according to https://github.com/docker/metadata-action#latest-tag ?

**Review by gerardsn — COMMENTED** [review]
the `v5` tag had the same issue (but probably limited to releases starting with v5). Can we apply the same fix there?

File: .github/workflows/build-images.yaml (hunk 1 above)
Comment (woutslakhorst) [reply .github/workflows/build-images.yaml:55]: yes and auto is broken, so now with scripting set to false/true

**Comment by woutslakhorst** [comment]
> the `v5` tag had the same issue (but probably limited to releases starting with v5). Can we apply the same fix there?

Not supported

**Comment by gerardsn** [comment]
so delete `v5`?

**Comment by woutslakhorst** [comment]
I'll merge for now since this is better then it was. Will address `v5` version later

**Pull request merged**